File: app/src/main/java/com/ytempest/wanandroid/activity/main/project/IProjectContract.java
package com.ytempest.wanandroid.activity.main.project;

import com.ytempest.wanandroid.base.presenter.IPresenter;
import com.ytempest.wanandroid.base.view.IView;
import com.ytempest.wanandroid.http.bean.ProjectClassifyBean;

import java.util.List;

/**
 * @author heqidu
 * @since 2020/6/30
 */
public interface IProjectContract {
    interface View extends IView {

        void onProjectClassifyReceived(List<ProjectClassifyBean> list);

        void onProjectClassifyFail(int code);
    }

    interface Presenter extends IPresenter {
        void getProjectClassify();
    }
}

File: app/src/main/java/com/ytempest/wanandroid/activity/main/project/IProjectContract.kt
package com.ytempest.wanandroid.activity.main.project


import com.ytempest.wanandroid.base.presenter.IPresenter
import com.ytempest.wanandroid.base.view.IView
import com.ytempest.wanandroid.http.bean.ProjectClassifyBean

/**
 * @author heqidu
 * @since 21-2-10
 */
interface IProjectView : IView {
    fun onProjectClassifyReceived(list: List<ProjectClassifyBean>)
    fun onProjectClassifyFail(code: Int)
}

interface IProjectPresenter : IPresenter {
    fun getProjectClassify()
}